### PR TITLE
test(headless): more controller test refactor for rtk2

### PR DESCRIPTION
**Pull request by olamothe**
https://coveord.atlassian.net/browse/KIT-2955

### CODE REVIEW

**Review by louis-bompart — APPROVED** [review]

File: packages/headless/src/controllers/core/facets/range-facet/date-facet/headless-core-date-facet.test.ts [hunk 1]
```diff
@@ -52,12 +64,22 @@ describe('date facet', () => {
     initDateFacet();
   });
 
-  it('#initDateFacet throws an error when an manual range in the options is invalid', () => {
+  it('#initDateFacet validate manual range in the options', () => {
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/core/facets/range-facet/date-facet/headless-core-date-facet.test.ts:67]:
```suggestion
  it('#initDateFacet validates manual range in the options', () => {
```

File: packages/headless/src/controllers/core/facets/range-facet/date-facet/headless-core-date-facet.test.ts [hunk 2]
```diff
@@ -97,35 +122,31 @@ describe('date facet', () => {
     it('dispatches a toggleSelectDateFacetValue with the passed value', () => {
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/core/facets/range-facet/date-facet/headless-core-date-facet.test.ts:122]:
```suggestion
    it('dispatches an #executeToggleDateFacetSelect with the passed value', () => {
```

File: packages/headless/src/controllers/core/facets/range-facet/date-facet/headless-core-date-facet.test.ts [hunk 3]
```diff
@@ -97,35 +122,31 @@ describe('date facet', () => {
...
-      expect(engine.actions).toContainEqual(action);
+      expect(executeToggleDateFacetSelect).toHaveBeenCalledWith({
+        facetId,
+        selection: value,
+      });
     });
   });
 
   describe('#toggleExclude', () => {
     it('dispatches a toggleExcludeDateFacetValue with the passed value', () => {
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/core/facets/range-facet/date-facet/headless-core-date-facet.test.ts:133]:
```suggestion
    it('dispatches an #executeToggleDateFacetExclude with the passed value', () => {
```

File: packages/headless/src/controllers/core/facets/range-facet/date-facet/headless-core-date-facet.test.ts [hunk 4]
```diff
@@ -97,35 +122,31 @@ describe('date facet', () => {
...
-    it('dispatches a #toggleSelect action with the passed facet value', () => {
-      dateFacet.toggleSingleSelect(facetValue());
-
-      expect(engine.actions).toContainEqual(
-        toggleSelectDateFacetValue({facetId, selection: facetValue()})
-      );
-    });
-
-    it('dispatches #updateFacetOptions with #freezeFacetOrder true', () => {
+    it('dispatches a #executeToggleDateFacetSelect action with the passed facet value', () => {
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/core/facets/range-facet/date-facet/headless-core-date-facet.test.ts:144]:
```suggestion
    it('dispatches an #executeToggleDateFacetSelect action with the passed facet value', () => {
```

File: packages/headless/src/controllers/core/facets/range-facet/date-facet/headless-core-date-facet.test.ts [hunk 5]
```diff
@@ -134,11 +155,10 @@ describe('date facet', () => {
 
     testCommonToggleSingleSelect(facetValue);
 
-    it('dispatches a #deselectAllFacetValues action', () => {
+    it('dispatches a #executeToggleDateFacetSelect action', () => {
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/core/facets/range-facet/date-facet/headless-core-date-facet.test.ts:158]:
```suggestion
    it('dispatches an #executeToggleDateFacetSelect action', () => {
```

File: packages/headless/src/controllers/core/facets/range-facet/date-facet/headless-core-date-facet.test.ts [hunk 6]
```diff
@@ -154,41 +174,24 @@ describe('date facet', () => {
...
-
-      expect(engine.actions).not.toContainEqual(
-        deselectAllDateFacetValues(facetId)
-      );
+      expect(deselectAllDateFacetValues).not.toHaveBeenCalled();
     });
   });
 
   function testCommonToggleExcludeSelect(facetValue: () => DateFacetValue) {
     it('dispatches a #toggleExclude action with the passed facet value', () => {
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/core/facets/range-facet/date-facet/headless-core-date-facet.test.ts:182]:
```suggestion
    it('dispatches an #executeToggleDateFacetExclude with the passed facet value', () => {
```

**Review by fbeaudoincoveo — APPROVED** [review]
My comments are all pet peeves ;)

**Pull request merged**